Adopt ServiceExtensions ScrollView and Delegate.

megangardner opened the pull request:
#### e171a196292b735a205a9adede19a265f93fb4a7
<pre>
Adopt ServiceExtensions ScrollView and Delegate.
<a href="https://bugs.webkit.org/show_bug.cgi?id=266100">https://bugs.webkit.org/show_bug.cgi?id=266100</a>
<a href="https://rdar.apple.com/119398513">rdar://119398513</a>

Reviewed by Wenson Hsieh.

Adoption of scroll view in ServiceExtensions and the linking
of that framework.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKSEDefinitions.h: Added.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.h:
* Source/WebKit/UIProcess/ios/WKBaseScrollView.h:
* Source/WebKit/UIProcess/ios/WKScrollView.h:
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollViewDelegateForwarder initWithInternalDelegate:externalDelegate:]):
(-[WKScrollView _updateDelegate]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/271795@main">https://commits.webkit.org/271795@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32b2fe55153518382391990c0c7d9703133e6304

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29624 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8281 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30941 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32130 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26810 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10491 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5570 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26829 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29897 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6958 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25282 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5900 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6080 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33482 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27054 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26788 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32259 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5988 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4188 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30031 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7726 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7047 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6532 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6512 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -292,4 +292,14 @@ WK_TRACE_POINTS_OUTPUT_PATH = $(WK_TRACE_POINTS_OUTPUT_PATH_USE_OVERRIDE_FRAMEWO
 WK_TRACE_POINTS_OUTPUT_PATH_USE_OVERRIDE_FRAMEWORKS_DIR_NO_macosx = $(DSTROOT)/AppleInternal/Library/Ariadne/Plists/WebKit.plist;
 WK_TRACE_POINTS_OUTPUT_PATH_USE_OVERRIDE_FRAMEWORKS_DIR_NO_iphoneos = $(DSTROOT)/AppleInternal/Library/Ariadne/Plists/WebKit.plist;
 
+WK_SERVICE_EXTENSIONS_LDFLAGS =
```
Nit - it looks like the other linker flag settings are defined before they're used in the definition of `FRAMEWORK_AND_LIBRARY_LDFLAGS`. Can we move this to before `FRAMEWORK_AND_LIBRARY_LDFLAGS` for consistency?

whsieh commented on Source/WebKit/UIProcess/API/ios/WKSEDefinitions.h:
```diff
@@ -0,0 +1,39 @@
+
+#if USE(APPLE_INTERNAL_SDK)
+
+#include <WebKitAdditions/WKSEDefinitionsAdditions.h>
```
Nit - `#import`

whsieh commented on Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.h:
```diff
@@ -25,11 +25,13 @@
 #import <WebCore/ScrollingTreeScrollingNode.h>
 #import <WebCore/ScrollingTreeScrollingNodeDelegate.h>
 
+
```
Nit - stray newline.

megangardner committed:
Adopt ServiceExtensions ScrollView and Delegate.
https://bugs.webkit.org/show_bug.cgi?id=266100
rdar://119398513

Reviewed by Wenson Hsieh.

Adoption of scroll view in ServiceExtensions and the linking
of that framework.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKSEDefinitions.h: Added.
* Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.h:
* Source/WebKit/UIProcess/ios/WKBaseScrollView.h:
* Source/WebKit/UIProcess/ios/WKScrollView.h:
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollViewDelegateForwarder initWithInternalDelegate:externalDelegate:]):
(-[WKScrollView _updateDelegate]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/271795@main
